Sources/StreamChat/Database/DatabaseSession.swift

```diff
@@ -399,13 +399,18 @@ extension DatabaseSession {
             // Message does not exits locally and should not be saved
             return
         }
-        
+
         let savedMessage = try saveMessage(
             payload: messagePayload,
             channelDTO: channelDTO,
             syncOwnReactions: false
         )
 
+        if payload.eventType == .messageDeleted && payload.hardDelete {
+            delete(message: savedMessage)
+            return
+        }
```

Comment:
Does it work fine for quote/thread replies when the parent message is deleted?

Reply:
What do you mean? 🤔 If the parent message is deleted, or if a reply is deleted, what happens?